### DISCUSSION TITLE
Planning of DOS port of the game

### DISCUSSION THREAD

**Discussion by Wohlstand** [Wohlstand/TheXTech#729] [Planning]
For a long time, I do have various ideas to make the game run on the DOS computer just for a fun. I actually own a very old computer with 486DX2 66 Hz processor and 32 MB RAM, and I am curious how much the game is able to work on such weak machinge.

Actually, there are next cases needed to be optimized manually:
- Need a totally custom graphical engine that will use various tweaks (source codes of Doom / Duke Nukem 3D / etc. can be used for getting of hints on how to deal with graphics on the low level).
- Obviously, all resources of episodes and the thing itself must be converted into special format with a very low colour depth (probably just up to 256). The format similar to BMP, but with a transparency.
- It's need to use the DOS Extender thing, so, it's possible to build the stuff using DJGPP already. I do already have an experience to use the DJGPP as a cross-compiler from the Linux, and I actually use it to build the stuff for DOS using it.
- Sound can be simplified if being used with native libraries such as AIL32 or Apogee Sound System ([here](https://github.com/FrenkelS/djdoom) is a possible thing that builds on DJGPP, and it's need to test it out in action by the way).
- Because of the 8.3 file system limitation, it's will be required to PACK all resources to the package. So, this depends on the planned archived format for games and episodes.

**Comment by LoveBodhi** [comment]
Some tests will be required:

- Under FreeDOS
- with Nostalgic Paradise (that will be non-SMBX64 objects in the future!)
